thewalrus/quantum/__init__.py

```diff
@@ -0,0 +1,225 @@
+
+Adjacency matrices
+^^^^^^^^^^^^^^^^^^
+
+.. autosummary::
+
+    adj_scaling
+    adj_scaling_torontonian
+    adj_to_qmat
+
+Gaussian checks
+^^^^^^^^^^^^^^^
+
+.. autosummary::
+
+    is_valid_cov
+    is_pure_cov
+    is_classical_cov
+    fidelity
+
+Conversions
+^^^^^^^^^^^
+
+.. autosummary::
+
+    reduced_gaussian
+    Xmat
+    Qmat
+    Covmat
+    Amat
+    complex_to_real_displacements
+    real_to_complex_displacements
+
+Means and variances
+^^^^^^^^^^^^^^^^^^^
+
+.. autosummary::
+
+    photon_number_mean
+    photon_number_mean_vector
+    photon_number_covar
+    photon_number_covmat
+    photon_number_expectation
+    photon_number_squared_expectation
+    normal_ordered_expectation
+    mean_clicks
+    variance_clicks
+
+Photon number distributions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. autosummary::
+
+    pure_state_distribution
+
+Details
+^^^^^^^
+"""
+import warnings
+
+from .fock_tensors import (
+    pure_state_amplitude,
+    state_vector,
+    density_matrix_element,
+    density_matrix,
+    fock_tensor,
+    probabilities,
+    loss_mat,
+    update_probabilities_with_loss,
+    update_probabilities_with_noise,
+    find_classical_subsystem,
+)
+
+from .adjacency_matrices import (
+    adj_scaling,
+    adj_scaling_torontonian,
+    adj_to_qmat,
+)
+
+from .gaussian_checks import (
+    is_valid_cov,
+    is_pure_cov,
+    is_classical_cov,
+    fidelity,
+)
+
+from .conversions import (
+    reduced_gaussian,
+    Xmat,
+    Qmat,
+    Covmat,
+    Amat,
+    complex_to_real_displacements,
+    real_to_complex_displacements,
+)
+
+from .means_and_variances import (
+    photon_number_mean,
+    photon_number_mean_vector,
+    photon_number_covar,
+    photon_number_covmat,
+    photon_number_expectation,
+    photon_number_squared_expectation,
+    normal_ordered_expectation,
+    mean_clicks,
+    variance_clicks,
+)
+
+from .photon_number_distributions import (
+    pure_state_distribution,
+)
+
+__all__ = [
+    "pure_state_amplitude",
+    "state_vector",
+    "density_matrix_element",
+    "density_matrix",
+    "fock_tensor",
+    "probabilities",
+    "loss_mat",
+    "update_probabilities_with_loss",
+    "update_probabilities_with_noise",
+    "find_classical_subsystem",
+    "adj_scaling",
+    "adj_scaling_torontonian",
+    "adj_to_qmat",
+    "is_valid_cov",
+    "is_pure_cov",
+    "is_classical_cov",
+    "fidelity",
+    "reduced_gaussian",
+    "Xmat",
+    "Qmat",
+    "Covmat",
+    "Amat",
+    "complex_to_real_displacements",
+    "real_to_complex_displacements",
+    "photon_number_mean",
+    "photon_number_mean_vector",
+    "photon_number_covar",
+    "photon_number_covmat",
+    "photon_number_expectation",
+    "photon_number_squared_expectation",
+    "normal_ordered_expectation",
+    "mean_clicks",
+    "variance_clicks",
+    "pure_state_distribution",
+]
+
+def deprecate(new_func):
+    """Wrapper for deprecated functions to raise warning"""
+    def wrapper(*args, **kwargs):
```

Comment:
It might be nice to use [`functools.wraps`](https://docs.python.org/3/library/functools.html#functools.wraps) here, so that the wrapped functions display the correct docstrings when in the terminal/a jupyter notebook:

```suggestion
def deprecate(new_func):
    """Wrapper for deprecated functions to raise warning"""
    
    @functools.wraps(new_func)
    def wrapper(*args, **kwargs):
```